jakarta.ls/src/main/java/org/eclipse/lsp4jakarta/JakartaTextDocumentService.java

```diff
@@ -156,15 +170,24 @@ private void triggerValidationFor(List<String> uris) {
         // javaParams.setDocumentFormat(DocumentFormat.Markdown);
         // }
         javaParams.setDocumentFormat(DocumentFormat.Markdown);
-        jakartaLanguageServer.getLanguageClient().getJavaDiagnostics(javaParams) //
-                .thenApply(diagnostics -> {
-                    if (diagnostics == null) {
-                        return null;
-                    }
-                    for (PublishDiagnosticsParams diagnostic : diagnostics) {
-                        jakartaLanguageServer.getLanguageClient().publishDiagnostics(diagnostic);
-                    }
-                    return null;
-                });
+
+        // Async thread to query the JDT LS ext for Jakarta EE diagnostics
+        CompletableFuture<List<PublishDiagnosticsParams>> getJakartaDiagnostics = CompletableFuture.supplyAsync(() -> {
+            try {
+                // Pass the JakartaDiagnosticsParams to IDE client, to be forwarded to the JDT
+                // LS ext
+                return jakartaLanguageServer.getLanguageClient().getJavaDiagnostics(javaParams).get();
+            } catch (Exception e) {
+                LOGGER.severe("Return request from client did not succeed: " + e.getMessage());
```

Comment:
A minor comment that it may be beneficial to distinguish the two new log messages by adding request type
I think i.e. `Return diagnostic request ...` or `Return completion request ...`

Reply:
Good idea, added in the request type